server/coordinator/node_picker.go

```diff
@@ -0,0 +1,42 @@
+// Copyright 2022 CeresDB Project Authors. Licensed under Apache-2.0.
+
+package coordinator
+
+import (
+	"context"
+	"crypto/rand"
+	"math/big"
+
+	"github.com/CeresDB/ceresmeta/server/cluster"
+	"github.com/pkg/errors"
+)
+
+type NodePicker interface {
+	PickNode(ctx context.Context, clusterName string) (cluster.RegisteredNode, error)
```

Comment:
The `NodePicker` doesn't have to know anything about the cluster, so how about put it as:
```suggestion
	PickNode(ctx context.Context, nodes []*cluster.RegisteredNode) (int, error)
```

After that, unit tests will be more simple.

Reply:
That's right, I thought about it again. The more important reason is that the scheduler should be scheduled based on the static topology of the current version instead of dynamically obtaining the latest metadata. Maybe `ShardPicker` also needs similar modifications.